go/merge/three_way_list.go

```diff
@@ -94,13 +94,26 @@ func canMerge(a, b types.List, aSplice, bSplice types.Splice) bool {
 	if aSplice != bSplice {
 		return false
 	}
-	// TODO: Add List.IterFrom() and use that to compare added values.
-	for i := uint64(0); i < aSplice.SpAdded; i++ {
-		if !a.Get(aSplice.SpFrom + i).Equals(b.Get(bSplice.SpFrom + i)) {
-			return false
-		}
+	aChan, bChan := make(chan types.Value), make(chan types.Value)
+	iterFrom := func(l types.List, s types.Splice, c chan<- types.Value) {
+		l.IterFrom(s.SpFrom, func(v types.Value, idx uint64) (stop bool) {
+			if i := idx - s.SpFrom; i < s.SpAdded {
+				c <- v
+				return false
+			}
+			return true
+		})
+		close(c)
+	}
+	go iterFrom(a, aSplice, aChan)
+	go iterFrom(b, bSplice, bChan)
+
+	mergeable := true
+	for aVal := range aChan {
+		bVal := <-bChan
+		mergeable = mergeable && aVal.Equals(bVal)
```

Comment:
Ideally, but to do so I'd need to also have code to drain the channels and/or stop the iterations and THEN drain the channels.

It seems like a significant reduction in code clarity and brevity for an optimization that only helps in the case when we're about to bail on the merge anyway